app/src/androidTest/java/ch/epfl/skysync/viewmodel/FlightsViewModelTest.kt

```diff
@@ -154,6 +155,108 @@ class FlightsViewModelTest {
     assertEquals(TimeSlot.AM, viewModelAdmin.timeSlot)
   }
 
+  @Test
+  fun correctEquipmentModifyFlight() {
+    // Set up the ComposeTestRule with the necessary content
+    composeTestRule.setContent {
+      viewModelAdmin = FlightsViewModel.createViewModel(repository, dbSetup.admin1.id)
+      viewModelAdmin.setFlight(dbSetup.flight4)
+    }
+
+    runTest {
+      viewModelAdmin.refreshAvailableBalloons().join()
+      viewModelAdmin.refreshAvailableBaskets().join()
+      viewModelAdmin.refreshAvailableVehicles().join()
+      viewModelAdmin.refreshAvailableUsers().join()
+
+      var expectedBalloons =
+          listOf(dbSetup.balloon1, dbSetup.balloon2, dbSetup.balloon3).sortedBy { it.id }
+      var expectedBaskets =
+          listOf(dbSetup.basket1, dbSetup.basket2, dbSetup.basket3).sortedBy { it.id }
+      var expectedVehicles =
+          listOf(dbSetup.vehicle1, dbSetup.vehicle2, dbSetup.vehicle3).sortedBy { it.id }
+      var expectedUsers = listOf(dbSetup.pilot1, dbSetup.crew1, dbSetup.crew2).sortedBy { it.id }
+
+      val currentBalloons = viewModelAdmin.currentBalloons
+      val currentBaskets = viewModelAdmin.currentBaskets
+      val currentVehicles = viewModelAdmin.currentVehicles
+      val currentUsers = viewModelAdmin.availableUsers
+
+      assertEquals(expectedBalloons, currentBalloons.value.sortedBy { it.id })
+      assertEquals(expectedBaskets, currentBaskets.value.sortedBy { it.id })
+      assertEquals(expectedVehicles, currentVehicles.value.sortedBy { it.id })
+      assertEquals(expectedUsers, currentUsers.value.sortedBy { it.id })
+
+      viewModelAdmin.setDateAndTimeSlot(dbSetup.date1, TimeSlot.AM)
+
+      viewModelAdmin.refreshAvailableBalloons().join()
+      viewModelAdmin.refreshAvailableBaskets().join()
+      viewModelAdmin.refreshAvailableVehicles().join()
+      viewModelAdmin.refreshAvailableUsers().join()
```

Comment:
The refresh are not necessary as it is already done in setDateAndTimeSlot but you need to wait that these coroutines are finished (which is what your code does in a way).

Reply:
Yes I know. Do you have a better solution ?